src/Publisher/Factory.php

```diff
@@ -11,18 +11,18 @@
 class Factory
 {
     /**
-     * @param string $protocol
-     * @param string $host
-     * @param int $port
-     * @param string $path
-     * @param string $key
-     * @param string $secret
      * @param bool|string $hostname
-     * @param bool $ignoreSsl
-     * @return Publisher
      */
-    public function createPublisher($protocol, $host, $port, $path, $key, $secret, $hostname, $ignoreSsl)
-    {
+    public function createPublisher(
+        string $protocol,
+        string $host,
+        int $port,
+        string $path,
+        ?string $key,
+        ?string $secret,
+        $hostname,
+        bool $ignoreSsl = false
+    ): Publisher {
```

Comment:
This (and further edits) should normally be considered breaking changes. Are we considering to release this as a new major?

Reply:
> This (and further edits) should normally be considered breaking changes. Are we considering to release this as a new major?

I proposed this as a BC. I kept the 7.4 compatibility only to allow applying this upgrade before upgrading our project to PHP 8.

Reply:
@Jean85 yes, I'd propose to release those BCs with a new major.